### ISSUE TITLE
buggy placement of Dialog close button

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#346]
Example in plinko-probability:

``` js
new Dialog( new MultiLineText( 'Out\nof\nBalls!' ), { font: new PhetFont( 25 ) } ), {
  modal: true,
  focusable: true,
  xMargin: 40,
  yMargin: 20
} ).show();
```

Results in:

![screenshot_111](https://cloud.githubusercontent.com/assets/3046552/18110009/676c3abe-6ed2-11e6-969a-6d8f9a6c38c0.png)

Looks like the problem is here in Dialog:

``` js
126 closeButton.right = dialogContent.right + options.xMargin - options.closeButtonMargin;
127 closeButton.top = dialogContent.top - options.xMargin + options.closeButtonMargin;
```

I don't believe that the placement of the close button should have anything to do with the `dialogContent`. The close button should always be inside the dialog. And it should simply be placed in the upper-right corner of the dialog, with the margin specified by `options.closeButtonMargin`.

Assigning to @jonathanolson since he added this bit of code in @13b93ab2ed26820eb95553be3e9bdd0508536f05.

**Comment by pixelzoom** [comment]
Noted while working on https://github.com/phetsims/plinko-probability/issues/72.

**Comment by pixelzoom** [comment]
This problem seems to occur whenever you try to use the `xMargin` and `yMargin` options. I've worked around it in plinks-probability by avoiding those options, and wrapping my dialog in a sun.Panel with the desired margins.

**Comment by jonathanolson** [comment]
> I don't believe that the placement of the close button should have anything to do with the dialogContent

There's the complication that for dialog content that spans the entire width, we don't want the close button overlapping the content (thus either content would need to be pushed down, or the dialog would need to be wider).

I'd prefer to defer until Dialog design is completed (reassign if something is desired/needed before then).

**Comment by pixelzoom** [comment]
> I'd prefer to defer until Dialog design is completed 

Agreed.

**Comment by pixelzoom** [comment]
Moved this to #359.  Closing.

**Comment by pixelzoom** [comment]
A close button on dialogs is apparently required for a11y standards, see https://github.com/phetsims/joist/issues/413.  Raising the priority of this issue.

**Comment by pixelzoom** [comment]
This got moved to #413, but has not been addressed. Reopening because the "closed" state gives the illusion that this was fixed, and it is not.

**Comment by jonathanolson** [comment]
This actually looks buggy because xMargin is used for the vertical computation:
```js
closeButton.top = dialogContent.top - options.xMargin + options.closeButtonMargin;
```
Since by default xMargin and yMargin are the same, I believe this is how it snuck through.

I'll look into testing the fix to change it to yMargin (looks good in my use-case fix in pendulum-lab so far).

**Comment by jonathanolson** [comment]
It looks like we've been avoiding this, so my new usage that looked broken (where I noticed) is the only case where this even comes up. Will push fix.

**Comment by jonathanolson** [comment]
@jessegreenberg, you had recent work in the Dialog, do you mind verifying that this should be fine?

**Comment by jessegreenberg** [comment]
The change looks good, this does look like a pretty clear mistake.  I tested a bunch of Dialogs that are using yMargin including AboutDialog and UpdateDialog.  This also fixes the case in plinko-probability that opened this issue, when I remove the workaround added in https://github.com/phetsims/plinko-probability/commit/0ceac3b3ffa64c7f0c45e8ec9760ec2ab00475c4, the Dialog looks good:
![screen shot 2017-07-28 at 3 33 01 pm](https://user-images.githubusercontent.com/6396244/28737855-b1ccfdc4-73ad-11e7-812a-721ec525c5bb.png)

> The close button should always be inside the dialog.

This is always be true except when ```closeButtonMargin``` is negative. Closing.
